plugins/anchor-links/index.js

```diff
@@ -90,7 +103,14 @@ function processHeading(node, compatibilitySlug, links) {
     )} permalink">»</a>`,
   })
 
-  return node
+  const headingData = {
+    title,
+    level,
+    slug,
+    aliases,
+    permalinkSlug,
+  }
+  return [node, headingData]
```

Comment:
What's going on with the change of return type here?

Reply:
It's because `processHeading` gets all the data about a heading. Would you rather me pass the `headings` variable into this function as the fourth argument and have it push this `headingData` to it? I think that's also a reasonable approach!

Reply:
This seems reasonable to me 👍

Reply:
Updated!

Reply:
Ooh sorry I meant I thought your original approach seemed reasonable haha, but either one is fine 😄

Reply:
Oh, we'll see what @jescalan thinks! 😁

I since figured out today that `links` is passed into `processHeading` to be populated by other code, so I think it's reasonable to keep following that pattern by passing in `headings`.